lib/orbit-common/jsonapi/fetch-requests.js

```diff
@@ -0,0 +1,129 @@
+import { toArray } from 'orbit/lib/objects';
+import Transform from 'orbit/transform';
+import { QueryExpressionParseError } from '../lib/exceptions';
+
+function deserialize(source, data) {
+  const deserialized = source.serializer.deserialize(data);
+  const records = toArray(deserialized.primary);
+
+  if (deserialized.included) {
+    Array.prototype.push.apply(records, deserialized.included);
+  }
+
+  const operations = records.map(record => {
+    return {
+      op: 'replaceRecord',
+      record
+    };
+  });
+
+  return [Transform.from(operations)];
+}
+
+export const FetchRequestProcessors = {
+  recordsOfType(source, request) {
+    const { type } = request;
+    const hash = {};
+
+    if (request.filter) {
+      hash.data = hash.data || {};
+      hash.data.filter = request.filter;
+    }
+
+    return source.ajax(source.resourceURL(type), 'GET', hash)
+      .then(data => deserialize(source, data));
+  },
+
+  record(source, request) {
+    const { type, id } = request;
+
+    return source.ajax(source.resourceURL(type, id), 'GET')
+      .then(data => deserialize(source, data));
+  },
+
+  relationship(source, request) {
+    const { type, id, relationship } = request;
+
+    return source.ajax(source.resourceRelationshipURL(type, id, relationship), 'GET')
+      .then(raw => {
+        let relId = source.serializer.deserializeRelationship(raw.data);
+        return relId;
+      });
+  },
+
+  relatedRecords(source, request) {
+    const { type, id, relationship } = request;
+
+    return source.ajax(source.relatedResourceURL(type, id, relationship), 'GET')
+      .then(data => deserialize(source, data));
+  }
+};
+
+export function getFetchRequests(query) {
+  // For now, assume a 1:1 mapping between queries and fetch requests
+  return [getFetchRequest(query.expression)];
+}
+
+function getFetchRequest(expression) {
```

Comment:
Perhaps this could start with an empty request

```
getFetchRequest(expression, request = {})
```

with every expression mapped to a request modifier (rather than splitting out ExpressionToRequestMap and RequestModifierMap)